wasted-app/RestaurantRegistrationControl.cs

```diff
@@ -40,45 +39,53 @@ private bool checkIfTextBoxesAreFull()
             else return false;
         }
 
-        private void resetTextBoxes()
+        private void ResetTextBoxes()
         {
             restaurantNameTextBox.Text = "";
             latitudeTextBox.Text = "";
             longitudeTextBox.Text = "";
             mailTextBox.Text = "";
             passwordTextBox.Text = "";
             repeatPasswordTextBox.Text = "";
-            textBoxLostFocus("Restaurant name", restaurantNameTextBox);
-            textBoxLostFocus("Latitude", latitudeTextBox);
-            textBoxLostFocus("Longitude", longitudeTextBox);
-            textBoxLostFocus("Mail", mailTextBox);
-            textBoxLostFocus("Password", passwordTextBox);
-            textBoxLostFocus("Repeat Password", repeatPasswordTextBox);
+            TextBoxLostFocus("Restaurant name", restaurantNameTextBox);
+            TextBoxLostFocus("Latitude", latitudeTextBox);
+            TextBoxLostFocus("Longitude", longitudeTextBox);
+            TextBoxLostFocus("Mail", mailTextBox);
+            TextBoxLostFocus("Password", passwordTextBox);
+            TextBoxLostFocus("Repeat Password", repeatPasswordTextBox);
         }
 
         private void registerButton_Click(object sender, EventArgs e)
         {
-            if(passwordTextBox.Text == repeatPasswordTextBox.Text && checkIfTextBoxesAreFull())
+            if(passwordTextBox.Text == repeatPasswordTextBox.Text && CheckIfTextBoxesAreFull())
             {
+                ServicesController controller = ServicesController.Instance;
                 var mail = mailTextBox.Text;
                 var password = passwordTextBox.Text;
-                String credentialError = getValidationError(mail, password);
-                if (credentialError == "")
+                var restaurant = controller.RestaurantService.GetByMail(new Mail(mail));
+                if (restaurant == null)
                 {
-                    passwordError.Text = "";
-                    ServicesController controller = ServicesController.Instance;
-                    Credentials creds = new Credentials(mail, password);
-                    controller.RestaurantService.Register(creds, new Restaurant("todo", restaurantNameTextBox.Text, new Coords(Convert.ToDecimal(latitudeTextBox.Text), Convert.ToDecimal(longitudeTextBox.Text)), new Credentials()));
-                    MessageBox.Show("Registered successfully");
-                    goBack();
-                    resetTextBoxes();
+                    String credentialError = GetValidationError(mail, password);
```

Comment:
Maybe use var?

Reply:
We should define where to use var. I'd use it just on the ui code, because I believe that using explicit types on the backend add really helpful information (but that's just a personal preference). It is correct but we'll define the style